File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

char alphaShift(char c, int i);

int main(int argc, string argv[])
{
    //check if proper number or command line arguments were passed to main.  If not, display error message and return 1 to main.
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }

    //convert argument into ciper key
    int k =atoi(argv[1]);    
    //check that argument is a number or a number other than zero
    if (k == 0)
    {
        printf("Invalid ciper key\n");
        return 1;
    }
    
    //get plaintext from user
    printf("plaintext: ");
    string p = get_string();
    
    //iterate over each char in plaintext and encrypt it with alphaShift function written below
    for (int i = 0, n = strlen(p); i < n; i++)
    {
        
        //check to see if char is a-z or A-Z.  If not, do change it.
        if(isalpha(p[i]))
        {
            p[i] = alphaShift(p[i], k);
        }

    }
    
    //display encrypted text
    printf("ciphertext: %s", p);
    printf("\n");
    return 0;
}

char alphaShift(char c, int i)
{
    //initialize shift for lower case
    int x = 97;
    //switch shift for uppper case if needed
    if(isupper(c))
    {
        x = 65;
    }
    
    //shift char in the alphabet n number of spaces, where n equals int passed in
    c = x + ((c - x + i) % 26);
    return c;
}

File: caesar.py
#!/usr/bin/env python3

import cs50

from sys import argv

#define function to shift characters in the alphabet by a given number of places
def alpha_shift(c, i):
    #intialize for lower case
    x = 97
    #check for uppercase and adjust as needed
    if c.isupper():
        x = 65
    #get the unicode value of the character
    co = ord(c)
    #shift the unicode value by the number of places desired and change it back to a char
    c = chr(x + ((co - x + i) % 26))
    return c

def main():
    #validate proper number of arguments are passed in storing the cipher shift value if it is passed in
    if len(argv) != 2:
        print("Usage: ./caesar k")
        return 1
    else:
        k = int(argv[1])

    #get string from user
    text = cs50.get_string("plaintext: ")
    #setup new cipher version of the string
    cipher_text = ""
    #iterate over each char
    for char in text:
        #check for alpha characters
        if char.isalpha():
            #then shift only alpha chars based on cipher argument
            cipher_text += alpha_shift(char, k)
        else:
            #if not alpha char, then pass to cipher text unchanged
            cipher_text += char
    #print cipher text
    print("ciphertext: {}".format(cipher_text))

if __name__ == "__main__":
    main()